docs/learning/ECSqlReference/InstanceQuery.md

```diff
@@ -0,0 +1,285 @@
+
+## Accessing composite properties like `NavigationProperty`, `Point2d`, `Point3d` or `Struct`s
+
+Only top level instance property can be accessed via `$-><prop>` syntax. Doing something like `$->Model.Id` will not not work as of now. It might be supported in future but as of now any access-string within a composite property is not supported, if its the only property selected then zero row will be returned.
+
+Following type of properties can directly be use in filters and return strong type value.
+
+- DateTime
+- Integer
+- Long
+- Binary
+- String
+- Double
+
+Here is example of `RevitId` use with `IN()` clause.
+
+```sql
+    SELECT $ from [BisCore].[Element] WHERE $->RevitId In ( 1000, 2000, 3000 );
+```
+
+While composite properties are returned as `JSON`.
+
+```sql
+    SELECT $->[Model] from [RevitDynamic].[Computer] where [ECInstanceId] = 0x8000000014c;
+```
+
+above will return following
+
+| $ -> Model                                     |
+| ---------------------------------------------- |
+| `{"Id":"0x80000000003","RelECClassId":"0x51"}` |
+
+While following will not return any row
+
+```sql
+    SELECT $->[Model].[Id] from [RevitDynamic].[Computer] where [ECInstanceId] = 0x8000000014c;
+```
+
+But you can still do following to get child property
+
+```sql
+    SELECT JSON_EXTRACT($->[Model], '$.Id') AS ModelId from [RevitDynamic].[Computer] where [ECInstanceId] = 0x8000000014c;
+```
+
+above will return following
+
+| ModelId         |
+| --------------- |
+| `0x80000000003` |
+
+## Optional and non-optional instance properties
+
+By default, all properties accessed via instance accessor i.e. `$->prop` must exist in the class identifying the row for that row to qualify for output.
+
+If the user uses `?` after a property accessor e.g. `$->prop?` then it will be considered optional, and the row class will not be checked to see if the `prop` exists or not.
+
+The following query will return no row if there is no subclass of `Bis.Element` that has both properties `CodeValue` and `Foo` in it.
+
+```sql
+  SELECT ECClassId, ECInstanceId
+  FROM [BisCore].[Element]
+      WHERE $->CodeValue = 'Profiling' OR $->Foo = 'Hello'
+  LIMIT 1
+```
+
+On the other hand, the following query makes `Foo` optional by adding `?` at the end like `$->Foo?`. This will exclude this property from the list of instance properties that must exist in the class of a row for it to qualify for output.
+
+```sql
+  SELECT ECClassId, ECInstanceId
+  FROM [BisCore].[Element]
+      WHERE $->CodeValue = 'Profiling' OR $->Foo? = 'Hello'
+  LIMIT 1
+```
+
+> Note: Optional property may slow down performance while non-optional properties will improve the performance of instance query.
+
+## Accessing composite properties
+
+Only top level instance properties can be accessed using instance property accessor syntax `$-><prop>`.\
```

Comment:
Seems like the same information as above in `Accessing composite properties like `NavigationProperty`, `Point2d`, `Point3d` or `Struct`s`

Reply:
Removed